cvxpy/tests/test_constant_atoms.py

```
@@ -328,7 +331,7 @@ def get_indices(size):
     """Get indices for dimension.
     """
     if len(size) == 0:
-        return tuple()
```

Returning `tuple()` here did lead to all tests with `size = tuple()` being silently skipped.
Fixing this has caused some tests unrelated to the PR to fail. 

Some tests fail with errors which seemed to be discussed in older issues already:
```
>           raise TypeError("'h' must be a 'd' matrix of size (%d,1)" %cdim)
E           TypeError: 'h' must be a 'd' matrix of size (9,1)
```
Other tests failed for SCS, but passed for ECOS.
I was also surprised that the expected values for the `cp.norm()` needed to be updated. Were there any updates to the function which would make this expected, such as the default axis?

---

That's great you uncovered this issue! That's a lot of errors :( These constant_atoms tests can be surprisingly challenging for solvers. But the cvxopt failures are clearly an input formatting issue.

---

I have fixed most of the tests, the three cases below are still failing under CVXOPT only. Strangely, the tests pass for other solvers, and CVXOPT passes for other test cases of these atoms. Perhaps you or @rileyjmurray can give me a pointer for further investigation, as there seemed to be a discussion around this error in [issue 1147](https://github.com/cvxpy/cvxpy/issues/1147).
```py
    (cp.lambda_max, tuple(), [[[2, 0, 0], [0, 3, 0], [0, 0, 1]]], Constant([3])),
...
    (cp.matrix_frac, tuple(), [[1, 2, 3],
                               [[1, 0, 0],
                                [0, 1, 0],
                                [0, 0, 1]]], Constant([14])),
    (cp.matrix_frac, tuple(), [[1, 2, 3],
                               [[67, 78, 90],
                                [78, 94, 108],
                                [90, 108, 127]]], Constant([0.46557377049180271])),
```

---

Looking a bit further, I found that removing the following condition and keeping the `else` case only, all tests pass:
https://github.com/cvxpy/cvxpy/blob/8c1ca2901645497748dccfe65e3b7981bf9bbe7e/cvxpy/reductions/solvers/conic_solvers/cvxopt_conif.py#L123-L130

There is already an [open PR](https://github.com/cvxpy/cvxpy/pull/1250) on this block.

---

@phschiele I'll look at this tonight.

---

@phschiele can you change lines 123 -- 127 to the following?
```
        if len_eq >= A.shape[0]:
            # Then the given optimization problem has no conic constraints.
            # This is certainly a degenerate case, but we'll handle it downstream.
            data[s.G] = None
            data[s.H] = None
```
I believe that will fix things. Let me know if it doesn't! (I made the changes on a branch that's a little old.)

---

Many thanks, @rileyjmurray, all tests are passing after adding this fix.